### DISCUSSION TITLE
ERROR Error: [createInstance] e depends on UNKNOWN service IMarkerNavigationService.

### DISCUSSION THREAD

**Discussion by prathyukurra** [microsoft/monaco-editor#3759] [Q&A]
I am working with monaco editors 0.27.0  and using for JSON and angular 11. When ever I load editor page I see a strange error

ERROR Error: [createInstance] e depends on UNKNOWN service IMarkerNavigationService.

Error: [createInstance] e depends on UNKNOWN service IMarkerNavigationService.
    at Ko._createInstance (2-es2015.js:1)
    at Ko.createInstance (2-es2015.js:1)
    at new e (2-es2015.js:1)
    at new e (2-es2015.js:1)
    at new e (2-es2015.js:1)
    at 2-es2015.js:1
    at xr (2-es2015.js:1)
    at Object.create (2-es2015.js:1)
    at e.initMonaco (2-es2015.js:1)
    at e.ngAfterViewInit (2-es2015.js:1)
    at 2-es2015.js:1
    at l.invokeTask (polyfills-es2015.js:1)
    at Object.onInvokeTask (main-es2015.js:1)
    at l.invokeTask (polyfills-es2015.js:1)
    at a.runTask (polyfills-es2015.js:1)
    at invokeTask (polyfills-es2015.js:1)
    at invoke (polyfills-es2015.js:1)
    at n.args.<computed> (polyfills-es2015.js:1)

Can I know what is the issue

**Comment by alexdima** [comment]
@prathyukurra Could you please provide a repro case? Unfortunately I cannot reproduce when running from source, nor when testing with a packaged monaco editor version.

**Comment by vimal-m** [comment]
In my env, Angular 12.2.9 and monaco-editor 0.27.0, this issue is seen when the project is built with the production flag and not otherwise.
Let me see, how easy is it to reproduce on a fresh project.

**Comment by vimal-m** [comment]
> In my env, Angular 12.2.9 and monaco-editor 0.27.0, this issue is seen when the project is built with the production flag and not otherwise. Let me see, how easy is it to reproduce on a fresh project.

I could reproduce the issue in a fresh angular project.
Here is what I was doing
`import 'monaco-editor'`

and here is what I did to fix the issue
`import { editor } from 'monaco-editor/esm/vs/editor/editor.api';`

Hope this helps.

**Comment by prathyukurra** [comment]
@vimal-m  As You mentioned I updated import 
From:  import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api.js';

To : import * as editor from 'monaco-editor/esm/vs/editor/editor.api';
 
 But I still see the same issue

Attached error for reference



![image](https://user-images.githubusercontent.com/79115917/136608502-351b8114-4c70-4f09-a392-0dfa357182ff.png)
